blog-posts/boosting-developer-productivity-with-github-actions.md

```diff
@@ -0,0 +1,266 @@
+- "Manual events": You can manually trigger events by clicking a button in a workflow that says "Run workflow now"
+- "Scheduled events": You can also have scheduled events based on a cron job that will for example trigger a workflow once a week, every Tuesday morning or whenever you like.
+
+#### Runner
+There is one more thing missing to make this whole thing work and that is the actual Runner that runs your workflows. The runners are mere applications, running somewhere on a server, listening for your jobs to execute them.
+GitHub.com provides a lot of these Runners that you can use right away. With a free github account you can use up to 2000 minutes of runner time per month, which should be plenty enough for private projects so you can start experimenting with them right away.
+
+In case you need a very special environment or would like to have more control over the runner you can also host them yourself. With custom runners you can very specialized environments to run your Windows 95 jobs or that integrate particularly well with your existing infrastructure.
+
+Runners can have different labels so they can be easily targeted in a workflow. All publicly available runners have labels like `ubuntu-latest` or `windows-latest` to reveal the underlying operating system. There can also be other labels like `x64` or `gpu` to reveal information about the architecture.
+
+## Workflows
+With all that terminology out of the way, let's talk about workflows in more detail. Where are they stored, what do they look like from the inside and the outside?
+
+### Location of a GitHub Workflow
+Workflows are stored in your repository in a folder called `.github/workflows/` in the form of YAML files. Here you can see a schema of this:
+```sh
+.github
+├── ...
+└── workflows
+    ├── create-release.yml
+    ├── housekeeping.yml
+    ├── new-issues.yml
+    ├── pr-checks.yml
+    ├── release-comment-in-issues.yml
+    └── stale.yml
+...
+```
+
+When you add a `*.yml` file in that folder it will be picked up by GitHub automatically and will be added to the list of workflows in your repository. There is no additional configuration required.
+
+### Anatomy of a Workflow
+As already mentioned, Workflows are written in the YAML format. Let's look at a workflow that runs checks on Pull Requests:
+```yaml
+name: Pull Request checks (1)
+
+on: (2)
+  pull_request: (3)
+    types: [opened, reopened, edited, synchronize] (4)
+    branches: [main] (5)
+
+jobs: (6)
+  commitlint-pr-title:
+    name: Ensure proper PR title (7)
+    runs-on: ubuntu-latest (8)
+    steps: (9)
+      - uses: actions/checkout@v2 (10)
+
+      - uses: actions/setup-node@v2 (11)
+        with: (12)
+          node-version: '16' (13)
+
+      - name: manually install config-conventional commitlint preset (14)
+        run: npm install --no-save @commitlint/config-conventional (15)
```

Comment:
It looks like only `config-conventional` is installed, I guess the other 2 dependencies are installed as peer dependency?

Reply:
With "the other 2 dependencies" you are referring to the dependencies of the action, like `@actions/github` and `@actions/core` or to the other (3) `@commitlint/*` dependencies?
Then it's a No. These dependencies are bundled with the action so there is no need to install them. This particular dependency `@commitlint/config-conventional` is needed because the commitlint config that will be used by the action is extending `@commitlint/config-conventional` so it needs to be present on the runner. The action can not know which configs the specific repository's `commitlint` extends from so there is no way of pre-bundling all presets. 
As this has caused confusion earlier I'm considering removing this line completely and not talking about this... but I'd really like to keep it in here, so I'm open for suggestions to clarify this.